src/commands/reset.js

```diff
@@ -8,30 +10,76 @@ class ResetCommand extends BaseCommand {
   /**
    * @param {Object} args
    * @param {Object} flags
-   * @param {DockerCompose} dockerCompose
+   * @param {resetSystemConfig} resetSystemConfig
    * @param {Config} config
+   * @param {ConfigCollection} configCollection
+   * @param {DockerCompose} dockerCompose
+   * @param {Docker} docker
+   * @param {tenderdashInitTask} tenderdashInitTask
    * @return {Promise<void>}
    */
   async runWithDependencies(
     args,
     {
       verbose: isVerbose,
+      hard: isHardReset,
+      'platform-only': isPlatformOnlyReset,
     },
-    dockerCompose,
+    resetSystemConfig,
     config,
+    configCollection,
+    dockerCompose,
+    docker,
+    tenderdashInitTask,
   ) {
     const tasks = new Listr([
+      {
+        title: 'Stop services',
+        task: async () => dockerCompose.stop(config.toEnvs()),
+      },
+      {
+        title: 'Reset platform data',
+        enabled: () => isPlatformOnlyReset,
+        task: () => (
+          new Listr([
+            {
+              title: 'Remove platform services and associated data',
+              task: async () => dockerCompose.rmPlatformOnly(config.toEnvs()),
+            },
+            {
+              title: 'Clean up platform volumes',
+              task: async () => docker.pruneVolumes(),
+            },
+            {
+              title: 'Reset platform config',
+              enabled: () => isHardReset,
+              task: async () => resetSystemConfig(configCollection, config.getName(), true),
+            },
+          ])
+        ),
+      },
       {
         title: 'Reset node data',
+        enabled: () => !isPlatformOnlyReset,
         task: () => (
           new Listr([
             {
-              title: 'Remove Docker containers and associated data',
+              title: 'Remove all services and associated data',
               task: async () => dockerCompose.down(config.toEnvs()),
             },
+            {
+              title: 'Reset config',
+              enabled: () => isHardReset,
```

Comment:
You need to reset the config only in case if it’s system one.

Reply:
Added a new function to check if current config is system config and did some refactoring